### ISSUE TITLE
Enhancement: Be able to create new entity cloning existing one

### ISSUE THREAD

**Issue by sglebs** [jowilf/starlette-admin#313]
**Is your feature request related to a problem? Please describe.**
We need to create new records, just like existing ones but 3 or 4 fields need to be changed. It is very tedious to create a new entity and type everything (copy/paste). Especially in non-normalized legacy tables.

**Describe the solution you'd like**

I would like to be able to create a new entity/record, based on an existing one. I would like to just edit the fields that need to be changed.

**Describe alternatives you've considered**

Maybe one more icon, just like the "eye" and the "pencil", which then takes me to a CREATE screen, pre-populated with the cloned values?

Maybe select 1 with the checkbox and choose an action at the top, similar to delete? This approach is tacky because the user could have chosen N records, and this particular action only applies to 1 record. Maybe take the first one selected?

Maybe the VIEW screen could have a CLONE button? It would open a new screen where I would be editing a new entity, populated with the same values. Just the ID field would be different, of course.

**Additional context**

After having typed all options above, I think the most elegant is to have a CLONE button in the VIEW page. Perhaps between EDIT and DELETE.

**Comment by jowilf** [comment]
I think we can add a new button 'save as new' on the edit page, similar to Django-admin, which would create a new item instead of editing the existing one.

**Comment by hasansezertasan** [comment]
That would be a great feature.

**Comment by hasansezertasan** [comment]
Is this feature can be implemented with `row actions`?

**Comment by jowilf** [comment]
> Is this feature can be implemented with row actions?

Yes but I think it will be hard to find a general way to do this, especially with relationship models. Anyone who needs this could create a custom row actions
